pkg/docker/docker.go

```diff
@@ -41,13 +43,48 @@ func (c *Client) GetRunningContainers() ([]*ContainerInfo, error) {
 	}
 
 	for _, container := range containers {
-		log.Infof("GetRunningContainers: Discovered running docker: %s %s %s",
-			container.ID, container.Names[0], container.Labels["io.kubernetes.pod.uid"])
-		containerInfos = append(containerInfos,
-			&ContainerInfo{
+		log.Infof("GetRunningContainers: Discovered running docker: %s %s %s State: %s Status: %s ",
+			container.ID, container.Names[0], container.Labels["io.kubernetes.pod.uid"], container.State, container.Status)
+
+		containerType, ok := container.Labels["io.kubernetes.docker.type"]
+		if !ok {
+			log.Infof("GetRunningContainers: skip non pause container")
+			continue
+		}
+
+		if containerType != "podsandbox" {
+			log.Infof("GetRunningContainers: skip container type: %s", containerType)
+			continue
+		}
+
+		log.Debugf("GetRunningContainers: containerType %s", containerType)
+
+		if container.State != "running" {
+			log.Infof("GetRunningContainers: skip container who is not running")
+			continue
+		}
+
+		uid := container.Labels["io.kubernetes.pod.uid"]
+		_, ok = containerInfos[uid]
+		if !ok {
+			containerInfos[uid] = &ContainerInfo{
 				ID:     container.ID,
 				Name:   container.Names[0],
-				K8SUID: container.Labels["io.kubernetes.pod.uid"]})
+				K8SUID: uid}
+			continue
+		}
+
```

Comment:
Turn these into an if/else and a comment above about why you don't think that the code should reach here under normal circumstances.  Do we a theory on which of the containerInfos is the correct one in the case we have a collision on uuid but with different names?